nas_ports/security/pam_mkhomedir/files/patch-pam_mkhomedir.c

```diff
@@ -0,0 +1,17 @@
+--- pam_mkhomedir.c.orig	2016-05-25 08:06:06.324447000 -0500
++++ pam_mkhomedir.c	2016-05-25 08:09:46.686258000 -0500
+@@ -211,9 +211,11 @@
+ 		goto err;
+ 	}
+ 
+-	copymkdir(pwd->pw_dir, skeldir, getmode(set, S_IRWXU | S_IRWXG | S_IRWXO), pwd->pw_uid,pwd->pw_gid);
+-	free(set);
+-	return (PAM_SUCCESS);
++	if (strcmp(pwd->pw_dir,"/nonexistent") != 0 ) {
++	        copymkdir(pwd->pw_dir, skeldir, getmode(set, S_IRWXU | S_IRWXG | S_IRWXO), pwd->pw_uid,pwd->pw_gid);
++		free(set);
++		return (PAM_SUCCESS);
++	}
```

Comment:
I dont think this is right, this check should have been done way sooner in the function.
In your case it will return error if dir is nonexistent?

Line 173:

        if ( strcmp(pwd->pw_dir,"/nonexistent") == 0 || (stat(pwd->pw_dir,&st) == 0) && ( S_ISDIR(st.st_mode))  ) {
                return (PAM_SUCCESS);
        }